lib/usd/ui/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =======================================================================
+#
+
+cmake_minimum_required(VERSION 3.1.1)
+project(mayaUsdUI)
+
+if (POLICY CMP0074)
+    cmake_policy(SET CMP0074 OLD)
+endif()
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+set(CMAKE_PREFIX_PATH "${QT_LOCATION}")
+find_package(Qt5 COMPONENTS Core Gui Widgets REQUIRED)
+
+add_library(${PROJECT_NAME} SHARED
+	factories/TreeModelFactory.cpp
+	views/ItemDelegate.cpp
+	views/TreeHeader.cpp
+	views/TreeModel.cpp
+	views/TreeItem.cpp
+	views/USDImportDialog.cpp
+	views/USDSearchThread.cpp
+	views/USDImportDialog.ui
+	cmds/USDImportDialogCmd.cpp
+)
+
+list(APPEND mayaUsdUI_headers
+	factories/TreeModelFactory.h
+	views/ItemDelegate.h
+	views/TreeHeader.h
+	views/TreeModel.h
+	views/TreeItem.h
+	views/USDImportDialog.h
+	views/USDSearchThread.h
+	views/IUSDImportView.h
+	cmds/USDImportDialogCmd.h
+)
+
+# Promote headers
+mayaUsd_promoteHeaderList(SUBDIR "ui" HEADERS ${mayaUsdUI_headers})
+
+target_compile_definitions(${PROJECT_NAME}
+	PRIVATE
+		$<$<BOOL:${IS_MACOSX}>:OSMac_>
+		MAYAUSD_UI_EXPORT
+)
+
+target_include_directories(${PROJECT_NAME}
+	PUBLIC
+		${MAYA_INCLUDE_DIRS}
+		${PXR_INCLUDE_DIRS}
+		${CMAKE_BINARY_DIR}/include
+)
+
+target_link_libraries(${PROJECT_NAME}
+	PRIVATE
+		mayaUsd
+		usd
+		${MAYA_OpenMaya_LIBRARY}
+		${MAYA_OpenMayaUI_LIBRARY}
+		Qt5::Core
+		Qt5::Gui
+		Qt5::Widgets
+)
+
+set(MAYAUSDUI_INSTALL_PATH ${CMAKE_INSTALL_PREFIX}/lib)
+
+if(IS_MACOSX OR IS_LINUX)
+    # rpath setup
+    mayaUsd_init_rpath(rpath "plugin")
+    if(WANT_USD_RELATIVE_PATH)
+        mayaUsd_add_rpath(rpath "../../USD/lib")
+    elseif(DEFINED PXR_USD_LOCATION)
+        mayaUsd_add_rpath(rpath "${PXR_USD_LOCATION}/lib")
+    endif()
+    mayaUsd_add_rpath(rpath ${MAYAUSDUI_INSTALL_PATH})
+    mayaUsd_install_rpath(rpath ${PROJECT_NAME})
+endif()
+
+if(IS_LINUX)
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--enable-new-dtags" PARENT_SCOPE)
+
+    if(WANT_USD_RELATIVE_PATH)
+        mayaUsd_add_rpath(rpath "../../USD/lib64")
+    endif()
+endif()
+
+install(
+	TARGETS ${PROJECT_NAME}
+	LIBRARY
+		DESTINATION ${MAYAUSDUI_INSTALL_PATH}
+	ARCHIVE
+		DESTINATION ${MAYAUSDUI_INSTALL_PATH}
+	RUNTIME
+		DESTINATION ${MAYAUSDUI_INSTALL_PATH}
+)
+if(IS_WINDOWS)
+	install(FILES $<TARGET_PDB_FILE:${PROJECT_NAME}> DESTINATION ${MAYAUSDUI_INSTALL_PATH} OPTIONAL)
+endif()
+
+install(FILES ${mayaUsdUI_headers}
+	DESTINATION ${CMAKE_INSTALL_PREFIX}/include/mayaUsd/ui/
+)
+
+#==============================================================================
+# Tests
+#==============================================================================
+if (BUILD_TESTS)
+	# A simple executable used to test the import dialog.
+	add_subdirectory(demo)
+endif()
```

Comment:
This builds a standalone executable that opens the import ui dialog. Useful during development.